pwndbg/commands/heap.py

```diff
@@ -629,28 +634,45 @@ def vis_heap_chunks(addr=None, count=None, naive=None):
 
     for c, stop in enumerate(chunk_delims):
         color_func = color_funcs[c % len(color_funcs)]
-
+        out_for_chunk = ''
+        # we store if this chunk is in a bin so that we can underline it
+        was_in_bin = False
         while cursor != stop:
             if printed % 2 == 0:
-                out += "\n0x%x" % cursor
+                out_for_chunk += "\n0x%x" % cursor
 
             cell = pwndbg.arch.unpack(pwndbg.memory.read(cursor, ptr_size))
             cell_hex = '\t0x{:0{n}x}'.format(cell, n=ptr_size*2)
 
-            out += color_func(cell_hex)
-            printed += 1
+            
 
+            prev_len_of_labels = len(labels)
             labels.extend(bin_labels(cursor, bin_collections))
+            # an address was in a bin when a label was added
+            if  len(labels) != prev_len_of_labels:
```

Comment:
nitpick: there's an extra space after `if`

Reply:
There are a few rogue spaces kicking about, not a big deal but a linter could help avoid them in future.

Reply:
Also what does this line do?